### PR TITLE
[MOB-1406] RN-A Support for Track In App Calls

### PR DESCRIPTION
**Pull request by Ayyanchira**
Add support for track in-app open, click, close and consume apis
Added RNInterableInternal file to access package private methods.

### CODE REVIEW

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java [hunk 1]
```diff
@@ -53,8 +57,84 @@ public void sampleMethod(String stringArgument, int numberArgument, Callback cal
...
+
+    private IterableInAppDeleteActionType getIterableDeleteActionTypeFromInteger(@Nullable Integer actionType) {
+        if (actionType == null || actionType >= IterableInAppCloseAction.values().length || actionType < 0) {
+            return null;
+        } else {
+            return IterableInAppDeleteActionType.values()[actionType];
+        }
+    }
+    //---------------------------------------------------------------------------------------
+    //endregion
```
Comment (vbabenkoru) [inline android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:139]: Move this to a separate file.

File: android/src/main/java/com/iterable/iterableapi/RNIterableInternal.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.iterable.iterableapi;
+
+public class RNIterableInternal {
+
+    public static String getEmail(){
+        return IterableApi.getInstance().getEmail();
```
Comment (tapashmajumder) [inline android/src/main/java/com/iterable/iterableapi/RNIterableInternal.java:6]: This should be using a promise/future somehow I think. Not sure about how this needs to be done for Android.

File: android/src/main/java/com/iterable/iterableapi/RNIterableInternal.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    }
+
+    public static IterableInAppMessage getMessageById(String messageId) {
+        return IterableApi.getInstance().getInAppManager().getMessageById(messageId);
+    }
+
+    public static void trackInAppClose(String messageId,  String clickedUrl,IterableInAppCloseAction closeAction, IterableInAppLocation location){
```
Comment (tapashmajumder) [inline android/src/main/java/com/iterable/iterableapi/RNIterableInternal.java:13]: nitpicking  spacing after commas are not consistent.

File: android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java [hunk 4]
```diff
@@ -53,8 +56,58 @@ public void sampleMethod(String stringArgument, int numberArgument, Callback cal
 
     @ReactMethod
     public void setUserId(@Nullable String userId) {
-        Log.d("ReactNative", "setUserId");
+        IterableLogger.d(TAG, "setUserId");
         IterableApi.getInstance().setUserId(userId);
     }
-    
+
+    //region Track APIs
```
Comment (tapashmajumder) [inline android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java:63]: Nitpicking: A space would look better after `//`

File: android/src/main/java/com/iterable/iterableapi/RNIterableInternal.java (hunk 2 above)
Comment (Ayyanchira) [reply android/src/main/java/com/iterable/iterableapi/RNIterableInternal.java:6]: Oh... its not actually related to this ticket. RNIterableInternal is intented to be just a helper class and its method will not be triggered directly by RN javascript side. And hence they will not be returning promise. getEmail() just happened to be the first package private method that I tested and I forgot to remove that. 

Will use this function when adding support to getEmail.

Comment (Ayyanchira) [reply android/src/main/java/com/iterable/iterableapi/RNIterableInternal.java]: removing getEmail to remove ambiguity.. Will add it when adding support to getEmail. :)

File: android/src/main/java/com/iterable/reactnative/Serialization.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    static IterableInAppLocation getIterableInAppLocationFromInteger(@Nullable Integer location) {
+        if (location == null || location >= IterableInAppLocation.values().length || location < 0) {
+            return null;
+        } else {
+            return IterableInAppLocation.values()[location];
+        }
+    }
+
+    static IterableInAppCloseAction getIterableInAppCloseSourceFromInteger(@Nullable Integer location) {
```
Comment (tapashmajumder) [inline android/src/main/java/com/iterable/reactnative/Serialization.java:19]: Where is `IterableInAppCloseAction` class defined? I don't see this in the list of files.

Comment (Ayyanchira) [reply android/src/main/java/com/iterable/reactnative/Serialization.java:19]: Its an enum defined in our IterableApi SDK like this
image

Comment (tapashmajumder) [reply android/src/main/java/com/iterable/reactnative/Serialization.java:19]: Sorry I must be missing something. Which java file is this in?

Comment (tapashmajumder) [reply android/src/main/java/com/iterable/reactnative/Serialization.java:19]: Is it possible that you forgot to check in the file?

Comment (tapashmajumder) [reply android/src/main/java/com/iterable/reactnative/Serialization.java:19]: Ah got it. Sorry.
I thought this was somewhere in RN sdk codebase.

**Pull request merged**